case_utils/case_validate/__init__.py

```diff
@@ -64,13 +65,189 @@ class NonExistentCDOConceptWarning(UserWarning):
+    :param ontology_graph: The ontology graph to use for validation.
+    :return: The set of concepts in the data graph that are not part of the CDO ontologies.
+    """
+    # Construct set of CDO concepts for data graph concept-existence review.
+    cdo_concepts: Set[rdflib.URIRef] = set()
+
+    for n_structural_class in [
+        NS_OWL.Class,
+        NS_OWL.AnnotationProperty,
+        NS_OWL.DatatypeProperty,
+        NS_OWL.ObjectProperty,
+        NS_RDFS.Datatype,
+        NS_SH.NodeShape,
+        NS_SH.PropertyShape,
+        NS_SH.Shape,
+    ]:
+        for ontology_triple in ontology_graph.triples(
+            (None, NS_RDF.type, n_structural_class)
+        ):
+            if not isinstance(ontology_triple[0], rdflib.URIRef):
+                continue
+            if concept_is_cdo_concept(ontology_triple[0]):
+                cdo_concepts.add(ontology_triple[0])
+    for n_ontology_predicate in [
+        NS_OWL.backwardCompatibleWith,
+        NS_OWL.imports,
+        NS_OWL.incompatibleWith,
+        NS_OWL.priorVersion,
+        NS_OWL.versionIRI,
+    ]:
+        for ontology_triple in ontology_graph.triples(
+            (None, n_ontology_predicate, None)
+        ):
+            assert isinstance(ontology_triple[0], rdflib.URIRef)
+            assert isinstance(ontology_triple[2], rdflib.URIRef)
+            cdo_concepts.add(ontology_triple[0])
+            cdo_concepts.add(ontology_triple[2])
+    for ontology_triple in ontology_graph.triples((None, NS_RDF.type, NS_OWL.Ontology)):
+        if not isinstance(ontology_triple[0], rdflib.URIRef):
+            continue
+        cdo_concepts.add(ontology_triple[0])
+
+    # Also load historical ontology and version IRIs.
+    ontology_and_version_iris_data = importlib.resources.read_text(
+        case_utils.ontology, "ontology_and_version_iris.txt"
+    )
+    for line in ontology_and_version_iris_data.split("\n"):
+        cleaned_line = line.strip()
+        if cleaned_line == "":
+            continue
+        cdo_concepts.add(rdflib.URIRef(cleaned_line))
+
+    data_cdo_concepts: Set[rdflib.URIRef] = set()
+    for data_triple in data_graph.triples((None, None, None)):
+        for data_triple_member in data_triple:
+            if isinstance(data_triple_member, rdflib.URIRef):
+                if concept_is_cdo_concept(data_triple_member):
+                    data_cdo_concepts.add(data_triple_member)
+            elif isinstance(data_triple_member, rdflib.Literal):
+                if isinstance(data_triple_member.datatype, rdflib.URIRef):
+                    if concept_is_cdo_concept(data_triple_member.datatype):
+                        data_cdo_concepts.add(data_triple_member.datatype)
+
+    return data_cdo_concepts - cdo_concepts
+
+
+def validate(
+    input_file: str,
+    case_version: Optional[str] = None,
+    supplemental_graphs: Optional[List[str]] = None,
+    abort_on_first: bool = False,
+    inference: Optional[str] = "none",
```

Comment:
I would prefer if these were unordered keyword arguments, rather than this syntax that looks like keywords but I ambiguously parse as either ordered arguments with default-values OR keyword arguments that all happen to have default values.

Can this signature instead become:

```python
def validate(
    input_file: str,
    *args: Any,
    case_version: Optional[str] = None,
    supplemental_graphs: Optional[List[str]] = None,
    abort_on_first: bool = False,
    inference: Optional[str] = "none",
    **kwargs: Any,
) -> ValidationResult:
```

This also maintains a parallel signature style (aside from return type) with `pyshacl.validate()`.  Except, the first argument in pySHACL is `data_graph: Union[GraphLike, str, bytes]`.

We can potentially save on some parameter reinvention here by just forwarding `kwargs` with `pyshacl.validate(..., **kwargs)`.  E.g. we wouldn't need to include `abort_on_first` or `inference` in this signature.

@kchason , would you be OK with adopting this argument-forwarding style?

Reply:
Yes, that works for me. Once we get some of the other cherry-picked items merged, I'll consolidate those into my branch and update this PR accordingly.

Reply:
This should be addressed in 4208a99eab30330d20d1037bab0f52f086944da6

Reply:
Confirmed, it's addressed, though I did make a style revision in how the arguments are forwarded in `8f957dc`.  Thanks!